ikenna-portfolio/portfolio/src/main/java/com/google/sps/servlets/CountServlet.java

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Metadata;
+import com.google.sps.data.Metadata.Search;
+import com.google.sps.database.CommentDatabase;
+import java.io.IOException;
+import java.util.*;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.lang.*; 
+
+/** Servlet that returns JSON content. */
+@WebServlet("/count")
+public class CountServlet extends HttpServlet {
+
+  private int count;
+  private int page;
+  private Search search;
+  
+  private CommentDatabase database;
+  private Metadata metadata;
+
+  @Override
+  public void init() {
+    database = new CommentDatabase();
+    metadata = new Metadata();
+    count = metadata.getCount();
+    page = metadata.getPage();
+    search = metadata.getSearch();
+    metadata.setMaxPages(database.getMaxPages(count));
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+    response.getWriter().println(getJson());
+  }
+
+  
+  /**
+   * This method creates metadata from the comments container
+   * that describes how the comments in the comments container
+   * should be displayed.
+   * @param request 
+   * @param response 
+  */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
The POST method are basically a GET with parameters. Consider consolidating this with GET and have all the parameters passed from UI instead of keeping a state in the servlet.

Reply:
I'm kinda blocked on how to consolidate the doGet and doPost. Throughout the program, I redirect the webpage with doPost (or else it gets redirected to a blank page automatically) then the javascript fetches the doGet when the page loads. Is there an alternative way I'm just completely missing?